toolkit/analysis_and_visualization/classification/assets/scripts/app.js

```diff
@@ -0,0 +1,572 @@
+function filterLabelTable(rowNum, Datavar) {
+    var labelSummaryFiltered = [];
+    var allEmpty = true;
+    // eslint-disable-next-line no-unused-vars
+    $("input[id ^= \"fl_\"]").each(function (element) {
+        var property = this.id.split("_")[1];
+        var value = this.value;
+        var compare = this.getAttribute("data-compare");
+        if (value) {
+            allEmpty = false;
+
+            labelSummary.forEach(function (label) {
+                var compareResult = false;
+
+                if (compare === "contains") {
+                    compareResult = label[property].toLowerCase().includes(value.toLowerCase());
+
+                } else {
+                    compareResult = label[property] == value;
+                }
+                if (compareResult) {
+                    labelSummaryFiltered.push(label);
+                }
+            });
+        }
+    });
+    if (allEmpty) {
+        loadLabelSummary(labelSummary);
+    } else {
+        loadLabelSummary(labelSummaryFiltered);
+    }
+}
+
+// eslint-disable-next-line no-unused-vars
+function clearLabelFilter() {
+    $("input[id ^= \"fl_\"]").val("");
+
+    loadLabelSummary(labelSummary);
+}
+
+// eslint-disable-next-line no-unused-vars
+function highlightRow(element) {
+    var parentRow = element.parentElement.parentElement;
+    parentRow.classList.toggle("highlight-row");
+
+}
+
+
+//Create copy of imageSummary
+var imageSummaryFiltered = JSON.parse(JSON.stringify(imageSummary));
+
+// eslint-disable-next-line no-unused-vars
+function clearResultFilter() {
+    $("input[id ^= \"fli_\"]").val("");
+    imageSummaryFiltered = JSON.parse(JSON.stringify(imageSummary));
+    loadImageResults(imageSummaryFiltered);
+    updateItems();
+}
+
+function filterResultTable(e) {
+    imageSummaryFiltered = [];
+    var allEmpty = true;
+    if (e) {
+
+        // eslint-disable-next-line no-unused-vars
+        var code = (e.keyCode ? e.keyCode : e.which);
+    }
+    var combinationOp = $("input:radio[name='filterType']:checked").val();
+    var notOp = $("#not-op").is(":checked");
+
+
+    var filters = {};
+
+    // eslint-disable-next-line no-unused-vars
+    $("input[id ^= \"fli_\"]").each(function (element) {
+        var allNames = this.id.split("_");
+        var property_index = null;
+        var prop_name_new = allNames[1];
+        if (allNames[2]) {
+            property_index = parseInt(allNames[2], 10);
+            prop_name_new = prop_name_new + "_" + property_index;
+        }
+        var value = this.value;
+        var op = $("#fli_op_" + prop_name_new).val();
+        if (value) {
+            allEmpty = false;
+            filters[this.id] = [];
+            filters[this.id].push(op);
+            filters[this.id].push(value);
+
+        }
+    });
+
+    imageSummaryFiltered = imageSummary.filter(function (item) {
+        var allTrue = true;
+        var anyTrue = false;
+
+        for (var key in filters) {
```

Comment:
Codacy found an issue: [The body of a for-in should be wrapped in an if statement to filter unwanted properties from the prototype.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=6397592)